tests/tests/test_basic_integration.py

```diff
@@ -52,6 +52,54 @@ def get_devauth_base_path(self):
 
 
 class BaseTestBasicIntegration(MenderTesting):
+    def do_test_artifact_name_rofs(self, env, valid_image_rofs_with_mender_conf):
+        """Upgrade a device with one R/O image verifying the artifact_name"""
+
+        mender_device = env.device
+        devauth = DeviceAuthV2(env.auth)
+        deploy = Deployments(env.auth, devauth)
+
+        # in the so-called clean image the artifact_info is "artifact_name=mender-image-clean\n"
+        # see: https://github.com/mendersoftware/mender-qa/pull/580/commits/a4c54e9ebf5879f0837789cbc7c74cb1102b2caa#diff-7031fcd5c934f225781febe446a34594a9a36b335c540379c0dd376a8f06ba28R263
+        # so first we expect it to be set to master, as this is what we
+        # have in the initial image, after update we expect it to change
+        expected_artifact_info = "artifact_name=mender-image-master\n"
+        artifact_info = mender_device.run(
+            "cat /etc/mender/artifact_info | sed -e '/artifact_name/q'"
+        )
+        logger.info("before update got %s" % artifact_info)
+        assert artifact_info == expected_artifact_info
+
+        mender_conf = mender_device.run("cat /etc/mender/mender.conf")
+        valid_image_rofs = valid_image_rofs_with_mender_conf(mender_conf)
+        if valid_image_rofs is None:
+            pytest.skip(
+                "Thud branch and older from Yocto does not have R/O rootfs support"
+            )
+
+        # Verify that partition is read-only as expected
+        mender_device.run("mount | fgrep 'on / ' | fgrep '(ro,'")
+
+        host_ip = env.get_virtual_network_host_ip()
+        update_image(
+            mender_device,
+            host_ip,
+            install_image=valid_image_rofs,
+            compression_type="gzip",
+            devauth=devauth,
+            deploy=deploy,
+        )
+
+        # after the update we expect the artifact info to be different
+        # ("artifact_name=mender-image-clean\n")
+        # since this is what is in the embedded image decompressed
+        # in the tests/run.sh
+        artifact_info = mender_device.run(
+            "cat /etc/mender/artifact_info | sed -e '/artifact_name/q'"
+        )
+        logger.info("after update got %s" % artifact_info)
+        assert expected_artifact_info != artifact_info
```

Comment:
Also, given that this test is almost identical to `test_double_update_rofs`, except for this check, can we just do the comparison there? In the interest of saving test time?

Reply:
those tests are identical -- in a sense that they check exactly the same for two different images rofs and regular one. does that make sense?
do you think we should also check for mender monitor and mender-gateway images? (we are not deploying them, so I thought that we do not have to)

Reply:
after we discussed privately, and after I understood, I is fixed.